backend/packages/Upgrade/src/loaders/typeormLoader.ts

```diff
@@ -25,53 +21,74 @@ export const typeormLoader: MicroframeworkLoader = async (settings: Microframewo
         database: env.db.database,
       },
-      replicas: [],
+      slaves: [],
     },
     synchronize: env.db.synchronize,
     logging: env.db.logging,
     maxQueryExecutionTime: env.db.maxQueryExecutionTime,
     entities: env.app.dirs.entities,
     migrations: env.app.dirs.migrations,
   };
-  // connection options:
-  let masterConnectionOptions = [];
-  let replicaConnectionOption = [];
-  const replicas = Object.keys(host_replicas).length;
 
+  const replicaConnectionOption = {
+    name: CONNECTION_NAME.REPLICA,
+    type: env.db.type, // See createConnection options for valid types
+    replication: {
+      master: {
+        host: '',
+        port: null,
+        username: '',
+        password: '',
+        database: '',
+      },
+      slaves: [],
+    },
+    synchronize: env.db.synchronize,
+    logging: env.db.logging,
+    maxQueryExecutionTime: env.db.maxQueryExecutionTime,
+    entities: env.app.dirs.entities,
+    migrations: env.app.dirs.migrations,
+  };
+  // number of read replicas:
+  const replicas = Object.keys(host_replicas).length;
+  const replica_hosts = [];
   if (replicas === 0) {
+    // if no read replica is defined, then we use master host for replica connection to handle export data
     /* tslint:disable:no-string-literal */
-    masterConnectionOptions = commonConnectionOptions['replication']['replicas'] = [];
-    /* tslint:disable:no-string-literal */
-    replicaConnectionOption = commonConnectionOptions['replication']['replicas'] = [];
-  } else {
-    if (replicas === 1) {
-      /* tslint:disable:no-string-literal */
-      masterConnectionOptions = commonConnectionOptions['replication']['replicas'] = [];
-    } else {
-      const replica_hosts = [];
-      for (let i = 1; i < replicas; i++) {
-        replica_hosts.push({
-          host: host_replicas[i],
-          port: env.db.port,
-          username: env.db.username,
-          password: env.db.password,
-          database: env.db.database,
-        });
-      }
-      /* tslint:disable:no-string-literal */
-      masterConnectionOptions = commonConnectionOptions['replication']['replicas'] = replica_hosts;
-    }
+    replicaConnectionOption['replication']['master'] = {
+      host: env.db.host,
+      port: env.db.port,
+      username: env.db.username,
+      password: env.db.password,
+      database: env.db.database,
+    };
+  } else if (replicas >= 1) {
+    // if a single read replica is defined, then we use the first read replica host to handle export data
     /* tslint:disable:no-string-literal */
-    replicaConnectionOption = commonConnectionOptions['replication']['replicas'] = [{
+    replicaConnectionOption['replication']['slaves'] = [{
       host: host_replicas[0],
       port: env.db.port,
       username: env.db.username,
       password: env.db.password,
       database: env.db.database,
     }];
-  }
 
-  replicaConnectionOption['name'] = CONNECTION_NAME.REPLICA;
+    for (let i = 1; i < replicas; i++) {
```

Comment:
this array of replica hosts can be created by mapping over the hostnames array; if it's created at the top of the file it could be easier to manage how these objects are being inserted into the connection options.

Reply:
Yeah true.